src/netguard/tls_parser.c

```diff
@@ -180,7 +178,13 @@ static int parse_server_name_extension(const uint8_t *data, size_t data_len, cha
                 }
                 strncpy(hostname, (const char *)(data + pos + 3), len);
                 (hostname)[len] = '\0';
-                return len;
+                if (is_valid_utf8(hostname)) {
```

Comment:
This is a check we had before that got lost when we re-wrote the TLS parsing.
We've seen some instances where the hostname is not UTF-8. The JNI interface (`is_domain_blocked`) uses `NewStringUTF` to pass the domain name to JVM, this is just to ensure we never get there with non UTF string.